### PR TITLE
Blog - Understanding Django Signals

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
Preview URL:
 https://63d2e1383d04ff0d917d6f9c--ephemeral-narwhal-3ac3b2.netlify.app

**Comment by msali123** [comment]
@balapriyac requested changes done

File: blog/_posts/2023-03-02-django-signals.md [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+- `sender`: the model class that just created the instance.
+- `instance`: another object's name created by the model method `__init__()`.
+- `raw`: a Boolean to detect whether the sender model is properly saved or not. It defaults to True.
+- `using`: used to state database aliases.
+- `update_fields`: a list of fields supplied to the model method `save()`. If no value is specified, it defaults to None.
+
+So we have a model `Profile` that needs `pre_save`  and `post_save` signals. This is just an illustration of how these signals can be used.
+
+~~~{.python caption="pre_save.py"}
+#models.py
```
[the remark this answers is not in the document]
Comment (jalletto) [reply blog/_posts/2023-03-02-django-signals.md]: @balapriyac This has caption "pre_save.py" but then this comment that implies the file is named "models.py"?

File: blog/_posts/2023-03-02-django-signals.md [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+~~~
+
+You'll observe that in response to a `pre_save` signal to the receiver decorator, the `update _profile()` receiver function modifies the property `to_receive_new_user`. Every time an instance is created before being saved to the database, the receiver function is called. Additionally, you'll see that the `created` Boolean keyword was not included in the receiver parameters. This is because an instance has not yet been created.
+
+#### `post_save`
+
+This signal is activated each time a model instance is saved. Except for the Boolean keyword `created`, this signal has all the same arguments as `pre_save.` Once an instance is created, the model will send the Boolean `created`  to the receiver.
+
+~~~{.python caption="post_save.py"}
+#models.py
```
[the remark this answers is not in the document]
Comment (jalletto) [reply blog/_posts/2023-03-02-django-signals.md]: Same here.

File: blog/_posts/2023-03-02-django-signals.md [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+We created the class `SendForgetPasswordEmail` which inherits from the `Thread` Django base class. To avoid thread blocking, it is necessary to use a separate thread for sending emails.
+
+We also override the `run()` method to send the email. This method comes from the Python `Thread` base class. It runs the thread whenever a `start()` method is called on the thread. The Python thread `run()` and `start()` methods can be referenced here, [link](https://docs.python.org/3/library/threading.html#threading.Thread.run).
+
+In addition, the `get_otp()` method returns a one-time password sent to the user's email. Note that `_otp` with a preceding underscore is considered private property. The proper way of accessing such properties outside the class is to create a "get" method that returns them. This is what the  `get_otp()` method does.
+
+Let's now add the `ForgetPassword` model class. Add the following by editing your `models.py` file in the program:
+
+~~~{.python caption="models.py"}
+#models.py
```
[the remark this answers is not in the document]
Comment (jalletto) [reply blog/_posts/2023-03-02-django-signals.md]: same here

File: blog/_posts/2023-03-02-django-signals.md [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+
+path('user/account/forgot-password', views.ForgotPasswordView.as_view(), \
+name="forgot-password"),
+
+~~~
+
+Let's also modify the `views.py` file for the endpoint incoming HTTP request.
+
+~~~{.python caption="views.py"}
+#views.py
```
[the remark this answers is not in the document]
Comment (jalletto) [reply blog/_posts/2023-03-02-django-signals.md]: Same here

**Comment by jalletto** [comment]
@balapriyac Looks great, just left a few comments.

**Comment by balapriyac** [comment]
> @balapriyac Looks great, just left a few comments.

@jalletto thank you. The author shared [this GitHub repo](https://github.com/Horlawhumy-dev/earthly-django-signals-article) with us when requested. I think the author mistakenly added models.py as a comment (for both the save and delete signals). Sorry for not catching this earlier.

**Review by balapriyac — CHANGES_REQUESTED** [review]

**Comment by msali123** [comment]
@balapriyac  @jalletto  done with the requested changes.

**Review by jalletto — APPROVED** [review]

**Pull request merged**